cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -581,16 +581,28 @@ class DuplicateMountDirValidator(Validator):
     Verify if there are duplicated mount dirs between shared storage and ephemeral volumes.
     """
 
-    def _validate(self, mount_dir_list):
-        duplicated_mount_dirs = _find_duplicate_params(mount_dir_list)
-        if duplicated_mount_dirs:
-            self._add_failure(
-                "Mount {0} {1} cannot be specified for multiple file systems".format(
-                    "directories" if len(duplicated_mount_dirs) > 1 else "directory",
-                    ", ".join(mount_dir for mount_dir in duplicated_mount_dirs),
-                ),
-                FailureLevel.ERROR,
-            )
+    def _validate(self, shared_storage_name_mount_dir_tuple_list, local_mount_dir_instance_types_dict):
```

Comment:
In this validation we do a two steps verification:
1. first verify that there is no clash among shared storages
2. then, verify that there is no clash between shared storages and local storages.
Did you do it this way because the goal was to provide distinct messages for shared v/s shared and shared v/s local?

Reply:
Yes, because of the different messages